assignments/arrays.js

```diff
@@ -63,34 +63,69 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
 
 // ==== Challenge 1 ====
 // The dealer can't recall the information for a car with an id of 33 on his lot. Help the dealer find out which car has an id of 33 by logging the car's year, make, and model in the console log provided to you below:
-console.log(`Car 33 is a *car year goes here* *car make goes here* *car model goes here*` );
+
+console.log(inventory[32]);
 
 
 
 // ==== Challenge 2 ====
 // The dealer needs the information on the last car in their inventory.  What is the make and model of the last car in the inventory?  Log the make and model into the console.
-let lastCar = 0;
-console.log();
+let lastCar = (inventory.length)-1;
+console.log(inventory[lastCar]);
 
 // ==== Challenge 3 ====
 // The marketing team wants the car models listed alphabetically on the website. Sort all the car model names into alphabetical order and log the results in the console
-let carModels = [];
-console.log();
+let carModels=[]
+for (i=0; i<inventory.length; i++) {
+    
+    carModels.push((inventory[i]['car_model']));
+    }
+console.log(carModels.sort());
+    
 
+    
 // ==== Challenge 4 ====
 // The accounting team needs all the years from every car on the lot. Create a new array from the dealer data containing only the car years and log the result in the console.
 let carYears = [];
-console.log();
+
+for (i=0; i<inventory.length; i++) {
+    
+    carYears.push( (inventory[i]['car_year']));
+  }
+  console.log(carYears);
+
 
 // ==== Challenge 5 ====
 // The car lot manager needs to find out how many cars are older than the year 2000. Using the carYears array you just created, find out how many cars were made before the year 2000 by populating the array oldCars and logging it's length.
-let oldCars =[];
-console.log(); 
+carYears.sort()
+console.log(carYears)
+let oldCars=[];
+for (i=0; i<=carYears.length; i++ ) {
+  
+ if ((carYears[i]) < 2000){
+   oldCars.push(carYears[i]);
+ }
+  console.log(oldCars)
+
+}
+
 
 // ==== Challenge 6 ====
 // A buyer is interested in seeing only BMW and Audi cars within the inventory.  Return an array that only contains BMW and Audi cars.  Once you have populated the BMWAndAudi array, use JSON.stringify() to show the results of the array in the console.
```

Comment:
Could have done this one without two loops. Your solution is definitely not wrong but it will be slower than:
```
let BMWAndAudi =[]; 
for (let i = 0; i < inventory.length; i++) {
  if (inventory[i].car_make == "BMW") {
    BMWAndAudi.push(inventory[i]);
  } else if(inventory[i].car_make == "Audi") {
    BMWAndAudi.push(inventory[i]);
  }
}
console.log("This is a list of all the BMW and Audi cars on the lot: " + JSON.stringify(BMWAndAudi));
```
Note that the challenge wanted you to turn this into a JSON object using ```JSON.stringify```.